src/main/java/core/basesyntax/impl/WritingToNewFileImpl.java

```diff
@@ -0,0 +1,32 @@
+package core.basesyntax.impl;
+
+import core.basesyntax.service.WritingToNewFile;
+import core.basesyntax.db.FruitShopDao;
+import core.basesyntax.service.CantWorkWithThisFileException;
+
+import java.io.BufferedWriter;
+import java.io.FileWriter;
+import java.io.IOException;
```

Comment:
The exception handling in the catch block is not following best practices. Instead of creating a custom exception that implies an inability to work with any file, consider using a more generic exception and providing a detailed message about the specific issue encountered.